### PR TITLE
Try to clarify documentation in re: Partitions

### PR DESCRIPTION
**Pull request by postilion**
This is a first attempt to clear up some confusion in regards to the
nameing of partitions and the use of the various "partition-name"
configuration directives in imapd.conf(5).

It's my goal to eliminate all use of common words like "default" or
"archive" in the names of partitions in samples and descriptive
language, lest people perceive such use to mean that those names
*must* be used.

For example, rather than using the following:
    defaultpartition: default
    partition-default: /var/spool/cyrus

we can use:
    defaultpartition: part1
    partition-part1: /var/spool/cyrus

The file /lib/imapoptions has been lightly modified in service to this
cause, and to correct typos in the definition of searchpartition-name
settings.

Note: There had been some trailing white space in /lib/imapoptions which my editor removed, which accounts for the last couple of diffs in that file.

### CODE REVIEW

File: docsrc/imap/reference/admin/locations.rst [hunk 1]
```diff
@@ -35,6 +35,55 @@ In general, Cyrus allows one to maintain several separate partitions for
...
+
+    # The Mail Spool Partitions
+    partition-atom: /var/spool/cyrus/fast/mail/atom/
+    partition-ntoz: /var/spool/cyrus/fast/mail/ntoz/
+
+    # The Metadata Partitions
+    metapartition-atom: /var/spool/cyrus/fast/meta/atom/
+    metapartition-ntoz: /var/spool/cyrus/fast/meta/ntoz/
+
+    # Backup Partition
```
Comment (elliefm) [inline docsrc/imap/reference/admin/locations.rst]: I think having a backup partition entry specified here is misleading.  This is for the backup service, which should generally not be on the same Cyrus instance as your regular mail partitions.

Also, the backup service is experimental in 3.0.x anyway, so I don't think we want to encourage cargo-culting of configuration for it.

**Comment by nicolan** [comment]
Looks ok to me. I'll leave it to @elliefm to provide final approval on her requested changes.

**Comment by elliefm** [comment]
Thanks!

Any thoughts on whether this update is suitable for back porting to the stable branch?  It seems reasonable to do so, but I can't remember if there's a good reason not to.  Maybe it could be confusing to people who set up their 3.0 system before it changed?  @postilion your input would be particularly appreciated here since of the three of us, you're the one who actually uses these docs as a Cyrus admin!

**Comment by hagedose** [comment]
FWIW, I'm in favor of back porting these changes. I don't think they will cause confusion, and they will help sysadmins who are yet to make the switch to 3.0.

**Comment by postilion** [comment]
@elliefm Yes, please do backport this to 3.0  I would do so, but given that my Git skills present a clear and present danger to the codebase, I think this is better handled by you. :-)

Seriously, I was totally confounded by the documentation on these subjects, which is particularly embarrassing given that I wrote (or at least massaged) much of it.  The language was just too confusing.  There's still more work to be done, I will note, to remove the use of "default" as an example default, or of "archive" as an example archive partition name.
-nic

**Comment by elliefm** [comment]
Doc changes have been back ported to 3.0 :)

**Pull request merged**